src/OrleansGetEventStore/GetEventStoreProvider.cs

```diff
@@ -0,0 +1,304 @@
+        {
+            serviceId = providerRuntime.ServiceId.ToString();
+
+            this.Name = name;
+            this.Log = providerRuntime.GetLogger(string.Format("LogViews.{0}.{1}", GetType().Name, id));
+
+            pageSize = config.GetIntProperty(PageSizeParameterName, 500);
+
+            var connectionString = config.GetProperty(HostParameterName, "localhost:1113");
+            var connectionStringParts = connectionString.Split(':');
+            var hostName = connectionStringParts[0];
+            var hostPort = int.Parse(connectionStringParts[1]);
+            var hostAddress = Dns.GetHostAddresses(hostName).First(a => a.AddressFamily == AddressFamily.InterNetwork);
+            
+            Log.Info($"Init Name={Name} ConnectionString={connectionString} Severity={Log.SeverityLevel}");
+
+            var username = config.GetProperty(UsernameParameterName, "admin");
+            var password = config.GetProperty(PasswordParameterName, "changeit");
+
+            var settings = ConnectionSettings.Create()
+                .KeepReconnecting().KeepRetrying()
+                .SetDefaultUserCredentials(new UserCredentials(username, password));
+
+            this.Connection = EventStoreConnection.Create(settings, new IPEndPoint(hostAddress, hostPort));
+            await this.Connection.ConnectAsync();
+
+            Log.Verbose("Successfully Connected");
+        }
+
+        public Task Close()
+        {
+            this.Connection.Close();
+            return TaskDone.Done;
+        }
+ 
+        private static readonly Dictionary<string,object> emptyCommitHeaders = new Dictionary<string, object>();
+
+        public async Task<bool> Append(string streamName, IEnumerable<KeyValuePair<Guid,object>> events, int? expectedVersion)
+        {
+            var eventsToSave = events.Select(e => ToEventData(e.Key, e.Value, emptyCommitHeaders)).ToArray();
+
+            if (eventsToSave.Length == 0)
+                return true;
+
+            var version = expectedVersion.HasValue
+                ? expectedVersion.Value == 0
+                    ? ExpectedVersion.NoStream
+                    : expectedVersion.Value - 1
+                : ExpectedVersion.Any;
+
+            if (eventsToSave.Length < pageSize)
+            {
+                try
+                {
+                    await this.Connection.AppendToStreamAsync(streamName, version, eventsToSave);
+                }
+                catch (WrongExpectedVersionException)
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                try
+                {
+                    var transaction = await this.Connection.StartTransactionAsync(streamName, version);
+
+                    var position = 0;
+                    while (position < eventsToSave.Length)
+                    {
+                        var pageEvents = eventsToSave.Skip(position).Take(pageSize);
```

Comment:
I don't think it is common. Is Skip() really slow, even given that we apply it to an array? Seems like LINQ could be smarter than that. Maybe not.